packages/core/content-manager/server/src/controllers/collection-types.ts

```diff
@@ -598,18 +598,32 @@ export default {
       return ctx.forbidden();
     }
 
-    // TODO: fix
     const permissionQuery = await permissionChecker.sanitizedQuery.delete(query);
+    const populate = await getService('populate-builder')(model)
+      .populateFromQuery(permissionQuery)
+      .build();
 
-    const idsWhereClause = { id: { $in: ids } };
-    const params = {
-      ...permissionQuery,
-      filters: {
-        $and: [idsWhereClause].concat(permissionQuery.filters || []),
-      },
-    };
+    const { locale } = getDocumentLocaleAndStatus(body);
+
+    const documentLocales = await documentManager.findLocales(documentIds, model, {
+      populate,
+      locale,
+    });
```

Comment:
If you are only deleting one locale at a time, is it necessary to get multiple document locales here?

Side comment, you might need to update the typing of findLocales to accept an array of strings.

Reply:
I'm trying to think in future here 🤔 to when we need to implement the bulkDelete for multiple locales

But right now it works just like a `findMany`, do you think makes sense to keep it with findLocales or is better use findMany? (I guess is the same at the end)

Reply:
I don't have strong opnions here. I normally try to "avoid" implementing before any feature has been decided, to reduce complexity in code, but here it is just chanigng findMany to findLocales so I don't see any issue on using this  :)